_posts/2023-10-25-aws-lambda-linode-migrate.md

```diff
@@ -0,0 +1,383 @@
+
+```bash
+
+faas-cli secret create video-preview-s3-key \
+  --from-file .secrets/video-preview-s3-key
+
+faas-cli secret create video-preview-s3-secret \
+  --from-file .secrets/video-preview-s3-secret
+```
+
+> You can checkout the documentation for more info on [how to use secrets within your functions](https://docs.openfaas.com/reference/secrets/).
+
+The function handler folder includes a `requirements.txt` file that was created while scaffolding the video-preview function from the `python-http` template. All Python packages the function code depends on need to be added here. The `video-preview` function uses the official AWS SDK for python, `boto3` to upload files to any S3-compatible Object storage. The `ffmpeg-python` python package provides bindings to FFmpeg and is used to process the input video. Make sure both are included in the `requirements.txt` file:
+
+```
+boto3
+python-ffmpeg
+```
+
+Like with the AWS function you have to make sure all additional binaries and packages the code depends on are installed in the function image. In this case our code depends on FFmpeg. With the official python-http template the build argument, `ADDITIONAL_PACKAGE` can be used specify additional `[apk](https://wiki.alpinelinux.org/wiki/Alpine_Package_Keeper)` or `[apt](https://wiki.debian.org/AptCLI)` packages that need to be installed.
+
+Update the functions `stack.yml` configuration to include FFmpeg as an additional package:
+
+```diff
+ functions:
+   video-preview:
+     lang: python3-http
++    build_args:
++      ADDITIONAL_PACKAGE: "ffmpeg"
+```
+
+> See the docs for more details on [adding native dependencies](https://docs.openfaas.com/languages/python/#native-dependencies) to OpenFaaS Python functions.
+
+
+
+Configure your OpenFaaS function.
+
+Extract, transform, and load (ETL) jobs and content generation workflows such as creating PDF files or media transcoding require fast, scalable local storage to process large amounts of data quickly.
+
+OpenFaaS (highlight in case of media processing pipelines):
+- no hard memory limits for functions.
+- temporary files in `/tmp`, no hard limits on size. (Lambda used to be 512MB now up to 10GB configurable, at additional cost)
+
+
+## Taking it further
+
+We saw how to migrate a ETL workflow to S3. Linode Object storage does not support bucket notifications at the moment. Instead of having the functions triggered by an event we invoked it asynchronously with the name of the source file instead.
+
+Have a look at [Minio](https://min.io/) which supports bucket notification over HTTP. You could also run your own Ceph cluster on Linode with [Rook](https://rook.io/). At has supports for setting up S3 compatible Object storage and sending bucket notifications over HTTP.
+
+We saw how to deploy OpenFaaS in a managed Kubernetes cluster with LKE. Alternatively you can create a cluster yourself. Checkout our tutorial: [How to set up production-ready K3s with OpenFaaS with Akamai Cloud Computing]()
```

Comment:
A conclusion should be added with a call to action.

I'm also seeing a missing URL for a link in this section